### DISCUSSION TITLE
Availability of ActionView helpers in arb partial

### DISCUSSION THREAD

**Discussion by giuseb** [activeadmin/activeadmin#8588] [Help]
I have moved the content of a resource's `index` to an `.arb` partial:

``` ruby
index do
  render "my/custom/partial"
end
```

Things seem OK, except for text helpers.
For example, this fragment works well, as long as the arbre code is embedded in the resource file:

``` ruby
para class: "border-b text-gray-500" do
  pluralize(course.lessons.size, 'class')
end
```

But the output is an empty string if the same code is run in a partial.
Any suggestions?
Thanks!

**Comment by javierjulio** [comment]
@giuseb I believe you have to use `text_node` for that to work due to the differences between Arbre context and Rails helpers.

```ruby
para class: "border-b text-gray-500" do
  text_node pluralize(course.lessons.size, 'class')
end
```

**Comment by giuseb** [comment]
@javierjulio it does not seem to work, I am getting the same empty string.

BTW, if the `pluralize` method is not available in the context of an Arbre view, why am I not getting an error?
Is Arbre using `method_missing` or something like that?

**Comment by giuseb** [comment]
Please note: switching from an `.arb` partial to either `erb` or `haml` templating produced the desired result: text helpers work as expected. Unlike Arbre, the other templates do not "know" about the collection object, so I have to explicitly pass it to the partial:

```ruby
  index do
    render "faculty/courses/course_cards", courses: courses
  end
```

and here's the haml partial:
```haml
- courses.each do |course|
  .border-b.text-gray-500=pluralize(course.lessons.size, 'class')
```
